lib/kitchen/driver/openstack.rb

```diff
@@ -303,20 +302,25 @@ def get_public_private_ips(server)
       end
 
       def get_ip(server)
-        unless config[:floating_ip].nil?
-          debug "Using floating ip: #{config[:floating_ip]}"
-          return config[:floating_ip]
-        end
-        if config[:openstack_network_name]
-          debug "Using configured net: #{config[:openstack_network_name]}"
-          return server.addresses[config[:openstack_network_name]].first['addr']
+        if config[:private_network_only] && config[:openstack_network_name]
+          update_server = compute.get_server_details(server[:server_id])  # refresh the server object with the stuff from the cloud.
+          pub = update_server[:body]['server']['addresses'][config[:openstack_network_name]][0]['addr']
+        else
+          unless config[:floating_ip].nil?
+            debug "Using floating ip: #{config[:floating_ip]}"
+            return config[:floating_ip]
+          end
+          if config[:openstack_network_name]
+            debug "Using configured net: #{config[:openstack_network_name]}"
+            return server.addresses[config[:openstack_network_name]].first['addr']
+          end
+          pub, priv = get_public_private_ips(server)
+          priv ||= server.ip_addresses unless pub
```

Comment:
It could be but in the `pry` sessions we had, we had access to both `server` and `state`. `server` gave us what we wanted so that's why we refresh off it on line 306.